labtool2.0/src/components/pages/CoursePage.js

```diff
@@ -88,35 +88,54 @@ export class CoursePage extends React.Component {
    * is a teacher or student on a course.
    */
   render() {
-    const createIndents = (data, siId) => {
-      const indents = []
+    const numberOfCodeReviews = Array.isArray(this.props.courseData.data) ? Math.max(...this.props.courseData.data.map(student => student.codeReviews.length)) : 0
 
-      for (var i = 0; i < this.props.selectedInstance.weekAmount; i++) {
+    const createIndents = (weeks, codeReviews, siId) => {
+      const indents = []
+      let i = 0
+      for (; i < this.props.selectedInstance.weekAmount; i++) {
         let pushattava = (
           <Table.Cell key={i}>
             <p>-</p>
           </Table.Cell>
         )
 
-        for (var j = 0; j < data.length; j++) {
-          if (i + 1 === data[j].weekNumber) {
+        for (var j = 0; j < weeks.length; j++) {
+          if (i + 1 === weeks[j].weekNumber) {
             pushattava = (
               <Table.Cell key={i}>
-                <p>{data[j].points}</p>
+                <p>{weeks[j].points}</p>
               </Table.Cell>
             )
           }
         }
         indents.push(pushattava)
       }
+      let ii = 0;
+      codeReviews.forEach(cr => {
+        indents.push(<Table.Cell key={i + ii}>
+            {cr.points !== null ? (<p>{cr.points}</p>) : (<p>-</p>)}
+          </Table.Cell>)
+        ii++
+      })
+      while (ii < numberOfCodeReviews) {
+        indents.push(<Table.Cell key={i+ii}>
+            <p>-</p>
+          </Table.Cell>)
+        ii++
+      }
       return indents
     }
 
     const createHeaders = () => {
       const headers = []
-      for (var i = 0; i < this.props.selectedInstance.weekAmount; i++) {
+      let i = 0
+      for (; i < this.props.selectedInstance.weekAmount; i++) {
```

Comment:
Could this be done in one line: for (let i; ....

Reply:
I don't know if it could, but I know it shouldn't. ```let i = 0``` is on its own line because i is used outside the for-loop.